build_tools/checks.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+repo_root="$(dirname "$0")/.."
+
+"$repo_root/build_tools/style.fish" --all --check
+
+cargo build --workspace --all-targets
+
+cargo test --no-default-features --workspace --all-targets
+cargo test --doc --workspace
+RUSTDOCFLAGS='-D warnings' cargo doc --workspace --all
+
+# TODO: add cachedir?
```

Comment:
last time I measured, adding a cachedir makes very little difference on runtime, at least for a single test